modeldb/base/deployment.yaml

```diff
@@ -0,0 +1,173 @@
+---
```

Comment:
Do we want to bundle in all deployments together? Or is it possible to do something similar to what was done for Pipelines?

Reply:
kustomize examples use similar naming (deployment.yaml, service.yaml) though I only see grouping in a few files and they're called resources.yaml. Combining by kind makes creating the application.yaml pretty easy. For tf-job-operator/base it would look like

```
apiVersion: app.k8s.io/v1beta1
kind: Application
metadata:
  name: "tf-job-operator-application"
spec:
  componentKinds:
  - group: apps
    kind: Deployment
  - group: core
    kind: ConfigMap
  - group: core
    kind: ServiceAccount
  - group: core
    kind: Service
```

[manifests/pipeline/api-service](https://github.com/kubeflow/manifests/tree/master/pipeline/api-service/base) looks similar to tf-job-operator. 
Putting the name in front of the kind might be problematic if we're using namePrefixes